media-api/app/controllers/MediaApi.scala

```diff
@@ -243,8 +247,21 @@ object MediaApi extends Controller with ArgoHelpers {
       relatedLabelsLink = getRelatedLabelsLink(searchParams)
       links = suggestedLabelsLink :: List(prevLink, nextLink, relatedLabelsLink).flatten
     } yield respondCollection(imageEntities, Some(searchParams.offset), Some(totalCount), links)
+
+    val searchParams = SearchParams(request)
+
+    SearchParams.validate(searchParams).fold(
+      // TODO: respondErrorCollection?
+      errors => Future.successful(respondError(UnprocessableEntity, InvalidUrlParams.errorKey,
+        // Annoyingly `NonEmptyList` and `IList` don't have `mkString`
+        errors.map(_.message).list.reduce(_+ ", " +_), List(searchLink))
```

Comment:
Yea, I was thinking of the client  - having to deal with multiple error responses would be annoying. I would argue multiple, as that can be the case.